interp/src/primitives/stateful.rs

```diff
@@ -66,92 +102,23 @@ impl Primitive for StdMultPipe {
-            //if left and right are the same as the interior left and right
-            // if (left.as_u64() == self.left.as_u64())
-            //     & (right.as_u64() == self.right.as_u64())
-            // {
-            //if self.cycle_count == 1 {
-            //and cycle_count == 1, return the product, and write product as update
             self.update = Some(
                 Value::from(left.as_u64() * right.as_u64(), self.width)
                     .unwrap(),
             );
-            // //reset cycle_count
-            // self.cycle_count = 0;
-            //return
-            return vec![
-                (
-                    ir::Id::from("out"),
-                    TimeLockedValue::new(
-                        (&Value::from(
-                            left.as_u64() * right.as_u64(),
-                            self.width,
-                        )
-                        .unwrap())
-                            .clone(),
-                        1,
-                        //Some(Value::from(0, self.width).unwrap()),
-                        Some(self.product.clone()),
-                    )
-                    .into(),
-                ),
-                (
-                    "done".into(),
-                    PulseValue::new(
-                        done_val.unwrap().clone(),
-                        Value::bit_high(),
-                        Value::bit_low(),
-                        1,
-                    )
-                    .into(),
-                ),
-            ];
-            // } else {
-            //     //else just increment cycle_count
-            //     self.cycle_count += 1;
-            //     // and return whatever was committed to [product]
-            //     // not a TLV
-            //     return vec![(
-            //         ir::Id::from("out"),
-            //         self.product.clone().into(),
-            //     )];
-            // }
-            // } else {
-            //     //else, left!=left and so on, restart (write these new left and right to interior left and right),
-            //     //set cycle_count to 1
-            //     self.cycle_count = 1;
-            //     self.left = Value::clone(left);
-            //     self.right = Value::clone(right);
-            //     // and return whatever was committed to [product]
-            //     return vec![(
-            //         ir::Id::from("out"),
-            //         self.product.clone().into(),
-            //     )];
-            // }
-        } else {
-            //if [go] is low, return whatever is in product
-            //this is not guaranteed to be meaningful
-            return vec![(
-                ir::Id::from("out"),
-                //     Value::from(0, self.width).unwrap().into(),
-                // )];
-                self.product.clone().into(),
-            )];
         }
+        //if go is low don't do anything (don't overwrite Update)
+        vec![]
     }
 
     fn reset(
```

Comment:
The reset should clear the internal state buffer. It shouldn't be the case that stuff in the pipeline can inadvertently pass between different groups, though to be fair pipelining remains underspecified within calyx

Reply:
Oh, makes sense! I wasn't sure when reset was used + what it was supposed to do